packages/hash/api/src/graphql/resolvers/ontology/property-type.ts

```diff
@@ -0,0 +1,113 @@
+import { ApolloError } from "apollo-server-express";
+import { AxiosError } from "axios";
+
+import {
+  PersistedPropertyType,
+  MutationCreatePropertyTypeArgs,
+  MutationUpdatePropertyTypeArgs,
+  QueryGetPropertyTypeArgs,
+  Resolver,
+} from "../../apiTypes.gen";
+import { GraphQLContext } from "../../context";
+import { PropertyTypeModel } from "../../../model";
+import { nilUuid } from "../../../model/util";
+import { propertyTypeModelToGQL } from "./model-mapping";
+
+export const createPropertyType: Resolver<
+  Promise<PersistedPropertyType>,
+  {},
+  GraphQLContext,
+  MutationCreatePropertyTypeArgs
+> = async (_, params, { dataSources }) => {
+  const { graphApi } = dataSources;
+  const { accountId, propertyType } = params;
+
+  const createdPropertyTypeModel = await PropertyTypeModel.create(graphApi, {
+    accountId,
+    schema: propertyType,
+  }).catch((err: AxiosError) => {
+    if (err.response?.status === 409) {
+      throw new ApolloError(
+        `Property type with the same URI already exists. [URI=${propertyType.$id}]`,
+        "CREATION_ERROR",
+      );
+    }
+
+    throw new ApolloError(`Couldn't create property type.`, "CREATION_ERROR");
+  });
+
+  return propertyTypeModelToGQL(createdPropertyTypeModel);
+};
+
+export const getAllLatestPropertyTypes: Resolver<
+  Promise<PersistedPropertyType[]>,
+  {},
+  GraphQLContext,
+  {}
+> = async (_, __, { dataSources }) => {
+  const { graphApi } = dataSources;
+
+  const allLatestPropertyTypeModels = await PropertyTypeModel.getAllLatest(
+    graphApi,
+    {
+      /** @todo Replace with User from the request */
+      accountId: nilUuid,
+    },
+  ).catch((err: AxiosError) => {
+    throw new ApolloError(`${err.response?.data}`, "GET_ALL_ERROR");
+  });
+
+  return allLatestPropertyTypeModels.map((propertyTypeModel) =>
+    propertyTypeModelToGQL(propertyTypeModel),
+  );
+};
+
+export const getPropertyType: Resolver<
+  Promise<PersistedPropertyType>,
+  {},
+  GraphQLContext,
+  QueryGetPropertyTypeArgs
+> = async (_, { propertyTypeVersionedUri }, { dataSources }) => {
+  const { graphApi } = dataSources;
+
+  const propertyTypeModel = await PropertyTypeModel.get(graphApi, {
+    versionedUri: propertyTypeVersionedUri,
+  }).catch((err: AxiosError) => {
+    throw new ApolloError(`${err.response?.data}`, "GET_ERROR");
+  });
+
+  return propertyTypeModelToGQL(propertyTypeModel);
+};
+
+export const updatePropertyType: Resolver<
+  Promise<PersistedPropertyType>,
+  {},
+  GraphQLContext,
+  MutationUpdatePropertyTypeArgs
+> = async (_, params, { dataSources }) => {
+  const { graphApi } = dataSources;
+  const { accountId, propertyType } = params;
+
+  const propertyTypeModel = await PropertyTypeModel.get(graphApi, {
```

Comment:
Self-request change: This is wrong, as the `$id` on the incoming type here doesn't not exist.

Reply:
Addressed in 4fc12287f531288fcca760b252876bf8fb6acc08

Reply:
And incrementing logic addressed in 002685b449802de9a52da1497002d4560a9a2b8c